[Android][fix] Remove JSONBundleConverter usage for passing manifest from Android to JS

bbarthec opened the pull request:
# Why

`Expo Go` is unable to load any project on Android now.
Recent change in app manifest shape is responsible for that, namely, new `config plugin system` (for `ncl` project this file: https://github.com/expo/expo/blob/master/apps/native-component-list/app.config.js is adding some array with strings/objects alternately).
That is causing https://github.com/expo/expo/blob/fec8b1534e8d27584a685c01a3a12ff432b3fbbd/android/expoview/src/main/java/host/exp/exponent/utils/JSONBundleConverter.java#L72-L81 to throw an error.

# How

There's no way to convert `JSONArray` with different entries to `Bundle` data type as the latter does not allow arrays/lists containing different data types, only homogenous entries are allowed.
I've removed this conversion altogether and replaced all `<someBundle>.putBundle("manifest", JSONBundleConverter.JSONToBundle(manifest))` with `<someBundle>.putString("manifest" manifest.toString())`.
I've tried to find any place in the code that would read the `"manifest"` fields from the bundle, but failed to do so 😞 cc @lukmccall, as there are changes in the `DevMenu` module.

From now on, `manifest` field in `initialProps` is replaced by `manifestString` that has to be deserialised upon access. Fortunately this field is not used anywhere in our codebase.

# Test Plan

- [x] I've succeeded in launching `ncl` and `test-suite` again on Android and observed no regressions. 
- [x] I've checked that DevMenu shows all the things in the header properly.
- [x] I've checked both Android & iOS behaviours.

# TODO

- [ ] Publish `dev-home` to accommodate deserialising manifest in order to read project's details (name/icon/etc).

bbarthec committed:
[Android] Remove JSONBundleConverter usage for passing manifest around

It is not possible to convert JSONArray that contains entries with differrent data types to Bundle.
The app manifest now contains some array with strings and objects alternately.
Manifest is now passed in stringified form to the bundle.

lukmccall commented:
Here you can find code that read the manifest: https://github.com/expo/expo/blob/bea037e227d332f2968d76effeb23fdc7b09b184/home/menu/DevMenuTaskInfo.tsx#L30-L35.

But if the dev menu header renders correctly (the icon, name, and others property are there), you can merge it if you ask me ;)

bbarthec committed:
[Android][iOS] Serialize & desarialize manifest JSON properly while passing it between JS & native

On Android the project manifest can no longer be converted into Bundle, and thus `manifest` field from `initialProps` (that are stored as Bundle) is now replaced by `manifestString` field that is deserialized on JS level.
On iOS side I've changed these fields similarly.

bbarthec committed:
[expo] Rebuild JS files
